internal/rukpak/api/bundle_types.go

```diff
@@ -0,0 +1,123 @@
+	// Directory refers to the location of the bundle within the git repository.
+	// Directory is optional and if not set defaults to ./manifests.
+	Directory string `json:"directory,omitempty"`
+	// Ref configures the git source to clone a specific branch, tag, or commit
+	// from the specified repo. Ref is required, and exactly one field within Ref
+	// is required. Setting more than one field or zero fields will result in an
+	// error.
+	Ref GitRef `json:"ref"`
+	// Auth configures the authorization method if necessary.
+	Auth Authorization `json:"auth,omitempty"`
+}
+
+type ConfigMapSource struct {
+	// ConfigMap is a reference to a configmap in the rukpak system namespace
+	ConfigMap corev1.LocalObjectReference `json:"configMap"`
+	// Path is the relative directory path within the bundle where the files
+	// from the configmap will be present when the bundle is unpacked.
+	Path string `json:"path,omitempty"`
+}
+
+type HTTPSource struct {
+	// URL is where the bundle contents is.
+	URL string `json:"url"`
+	// Auth configures the authorization method if necessary.
+	Auth Authorization `json:"auth,omitempty"`
+}
+
+type GitRef struct {
+	// Branch refers to the branch to checkout from the repository.
+	// The Branch should contain the bundle manifests in the specified directory.
+	Branch string `json:"branch,omitempty"`
+	// Tag refers to the tag to checkout from the repository.
+	// The Tag should contain the bundle manifests in the specified directory.
+	Tag string `json:"tag,omitempty"`
+	// Commit refers to the commit to checkout from the repository.
+	// The Commit should contain the bundle manifests in the specified directory.
+	Commit string `json:"commit,omitempty"`
+}
+
+type Authorization struct {
+	// Secret contains reference to the secret that has authorization information and is in the namespace that the provisioner is deployed.
+	// The secret is expected to contain `data.username` and `data.password` for the username and password, respectively for http(s) scheme.
+	// Refer to https://kubernetes.io/docs/concepts/configuration/secret/#basic-authentication-secret
+	// For the ssh authorization of the GitSource, the secret is expected to contain `data.ssh-privatekey` and `data.ssh-knownhosts` for the ssh privatekey and the host entry in the known_hosts file respectively.
+	// Refer to https://kubernetes.io/docs/concepts/configuration/secret/#ssh-authentication-secrets
+	Secret corev1.LocalObjectReference `json:"secret,omitempty"`
+	// InsecureSkipVerify controls whether a client verifies the server's certificate chain and host name. If InsecureSkipVerify
+	// is true, the clone operation will accept any certificate presented by the server and any host name in that
+	// certificate. In this mode, TLS is susceptible to machine-in-the-middle attacks unless custom verification is
+	// used. This should be used only for testing.
+	InsecureSkipVerify bool `json:"insecureSkipVerify,omitempty"`
+}
+
+type ProvisionerID string
```

Comment:
We don't need Provisioners any more, since we will have just one controller handling unpacking, and installing for all kinds of bundles. Also, for now, we are going to consider only registryV1 formats.